### DISCUSSION TITLE
Convert from .pth to .onnx

### DISCUSSION THREAD

**Discussion by Gurrman375** [rhasspy/piper#471] [Q&A]
I get my models from https://voice-models.com because of the selection they have. Is there anyway to convert these .pth and .index files to .onnx?

**Comment by synesthesiam** [comment]
Not that I know of. Are these for RVC?

**Reply by Gurrman375** [reply]
Yes they are RVC. I was looking into it and it seems that I might need a Nvidia GPU in order to convert.

**Comment by kvrban** [comment]
i would also be interested. i found something here. i don't know if it is the right one:
https://pytorch.org/tutorials/beginner/onnx/export_simple_model_to_onnx_tutorial.html

**Comment by MithrilMan** [comment]
any news on this?

**Comment by synesthesiam** [comment]
For this to work, the exported ONNX models would have to have the exact same input and output parameters as Piper's models.

**Reply by MithrilMan** [reply]
@synesthesiam I fear I've not the knowledge yet to fill the gaps, so let me try to summarize a bit my goal:
I found a repository ( https://github.com/dnhkng/GlaDOS/ ) that was using onnx model to play glados voice.
I got intrigued and tried to apply the same tech stack on C# net8 starting basically from 0 detailed knowledge.
I found onnxruntime, I used NAudio to get realtime audio and managed to make Silero VAD onnx model to work.
The original repo was using espeak to generate phonemes and after transforming them with the voice phonemesid found in the voice model config, it was passed as input to the VITS voice model. I've read that the repo owner used your training process to generate the voice.
Now, I was interested in being able to do multilanguage stuff (I'm Italian and wanted to use Italian voices) so I found a voice in piper that was working but the voice is quite bad, low quality and kind of boring for what I wanted (no other italian voice were available)
So I was looking for a way to generate easily some Italian voice to see how everything works together, before investing time in learning how to train and train a custom voice.
Since RVC2 had some interesting italian voices, I was looking for a way to convert these voices, hope this make sense.

Now, I don't want ready-made food, I like to learn and apply the knowledge so I'd be happy to understand better the insight, but at the same time I'm focused on what I'm building (for fun), kind of a real time personal assistant with funcion recall, etc... 
This AI space is running at the light speed, keeping the pace is impossible and being this a pet project (I'm not a youngster :D) I'd like to optimise my time and actually I haven't the skill to build models (but always willing to learn).
I don't know the difference between RVC(2) and VITS-piper, I'm not even sure RVC can work giving phonemes as input, what I like about piper is the inference speed that I can achieve but there is lack of voices.

Beside RVC, are you aware of other TTS that can be converted to piper easily?
If you have reference for learning purpose I'd be happy anyway to have a read.
Thanks from a builder :)

**Reply by xsploit** [reply]
i wish it was possible to convert sovits or rvc would make life easier
